connect/clients/ipfs.py

```diff
@@ -0,0 +1,146 @@
+"""
+ipfs.py
+
+IPFS client services that support REST calls for persistence of records with an
+IPFS cluster
+"""
+import logging
+import httpx
+import json
+from tempfile import NamedTemporaryFile
+from connect.config import get_settings
+
+
+logger = logging.getLogger(__name__)
+_ipfs_client = None
+
+
+# ******************************************
+# IPFS client services for persistence
+# ******************************************
+class IPFSClient:
+    def __init__(self, uri, replication_factor):
+        self._uri = uri
+        self._replication_factor = replication_factor
+        self._add_uri = self._uri + "/add"
+        self._get_peers_uri = self._uri + "/peers"
+        self._unpin_cid_uri = self._uri + "/pins/"
+
+    async def persist_json_to_ipfs(self, payload_dict):
+        """
+        Allows persistence of JSON payloads (as a dict) into an IPFS cluster
+        The IPFS cluster instance being connected to is configurable.
+
+        Content is pinned by default during the add operation.
+
+        :param payload_dict
+        :return content_cid: A unique ID addressing the content added to the
```

Comment:
May want to run a tuple here of (response_code, response) - that allows the caller of this method all of the relevant details if a 200 status code is not returned and leaves it up to the caller to parse out the relevant bits, in this case the "cid".  This may be a good pattern to consider for all the "client" methods in this module.